### PR TITLE
Security / PseudoCargo Expansion

### PR DESCRIPTION
**Pull request by Mechoid**
W.I.P., draft is here for easy referencing.

Behind-The-Scenes:
- Config entry added for enabling the New Event System.
- Basic cryopods added as ghost_pod types.

Additions:
 - Security can start with or order more NT-approved posters to put up around the station.
 - Modular armor components added to Cargo. They are now compatible with Old-style arm-guards and leg-guards, and cannot be stacked with said gear. [Old-style still covers hands / feet, modular don't.]
 - Magnetic weapons added to Cargo.
 - - HoS's E-gun has been replaced with a prototype magnetic pistol capable of firing slugs or stun-shells. Additionally, the only weapon to have a DNA lock. Cargo-ordered pistols do not have them.
 - - Sivian mag-rifles added to cargo. Projectiles function similarly to hunting rounds.
 - Detective bag added, for storing the Sheer Volumes of tiny items you acquire at a crime scene.
 - Detective scanner makes a return (from cargo) in a supporting role, providing limited information, such as only completed prints, blood DNA/type, and number of fibers. Can also take prints from a person.
 - Detective machinery operates faster.
 - Event added for exchanging cargo points for their value in thaler automatically, with the chance increasing as points pass certain thresholds. Thaler are added directly to the station account.
 - Event added to give Security a reason to give more fines, or for Cargo to provide sales, or for Civilian to actually charge.
 - Event added to give Security a reason to perform records upkeep / investigations.
 - Stowaway pods added as relatively rare events. [Observers used] In extended, they will only spawn Thugs, other modes can have Renegades or even full Traitors spawned.
 - Swarm Drones added as an Incredibly Rare event. [Observers used]
 - - Based on maintenance drones, however capable of local speech. Their unique language can be understood by translators.
 - - Basic Swarmer can produce one melee, and one ranged swarmer, but a functionally endless volume of standard swarmers. Ranged and Melee swarmers can only build regular swarmers.
 - - Make distinct use of the New pylon system, for special effects like shields and power supplies.
 - Beepsky and other bots move faster when the alert level is increased in response to a threat they handle. Medibots move faster in Violet, Floorbots move faster in Orange, Secbots move faster in Blue / Red, etc.
 - Spider event will now, when security is low, more than likely only spawn stunted spiderlings, which only grow into guardlings or hunterlings.

### CODE REVIEW

**Comment by zekesulastin** [comment]
Two questions about stowaway so they aren’t lost:

1) Isn’t a thug a higher “strength” antag than a renegade, spawngun aside?  The thug chat box blurb tells the thug to go rough people up etc. whereas the renegade blurb is entirely about self defense.
2) Since I’ve already shown I can’t really read code anymore, what bit keeps the full blown traitor from spawning in an Extended round? (I’m not entirely a fan of thugs either given their chat box blurb but that’s Zeke for ya)

Aside from that, does the manifest malfunction leave an error message telling people that it’s actually been corrupted instead of the default blank?  If not, I doubt people will think to check the paper records since blank is the most common state (also people will pitch an ic/ooc fit about cloning or medicine allergies or what have you).  Does it also have a chance to straight up delete people from the manifest as per the name?

Also, is there an easy way to make the security screening list only pick from things on the station?  It looks like the more heavily weighted things (like Zaddat) may not even be around to give low pop sec something to do.

**Comment by Mechoid** [comment]
The manifest malfunction deletes the respective record entirely. IIRC, removing someone from Sec records removes the manifest entry as well, hence the name.

Sec screening now respects actually present-in-records persons in order to gauge weight.

**Comment by Wickedtemp** [comment]
Tbh if its possible to have stowaways with injuries, like medivac stasis pods, could be neat for Medical.

File: code/modules/mob/living/silicon/robot/drone/swarm_items.dm [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+		else
+			to_chat(user, "<span class='notice'>There is already a disintigration field affecting \the [T].</span>")
+
+	if(grabbed_something)
+		to_chat(user, "<span class='notice'>You deploy your decompiler and clear out the contents of \the [T].</span>")
+	else
+		to_chat(user, "<span class='danger'>Nothing on \the [T] is useful to you.</span>")
+	return
+
+/obj/effect/temporary_effect/pulse/disintigrate
```
Comment (Anewbe) [inline code/modules/mob/living/silicon/robot/drone/swarm_items.dm]: > disintegrate

File: code/modules/mob/living/silicon/robot/drone/swarm_items.dm [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+				if(metal)
+					metal.add_charge(1000)
+		else
+			continue
+
+		qdel(W)
+		grabbed_something = 1
+
+	if(istype(T,/turf/simulated/wall) && (last_field < world.time + field_cooldown))
+		if(!(locate(/obj/effect/temporary_effect/pulse/disintigrate)))
```
Comment (Anewbe) [inline code/modules/mob/living/silicon/robot/drone/swarm_items.dm]: > disintegrate

File: code/modules/mob/living/silicon/robot/drone/swarm_items.dm [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+			last_field = world.time
+			to_chat(user, "<span class='alien'>You deploy an energetic field through \the [T], beginning its deconstruction.</span>")
+			to_chat(user, "<span class='warning'>You should stand back.</span>")
+			new /obj/effect/temporary_effect/pulse/disintigrate(T)
```
Comment (Anewbe) [inline code/modules/mob/living/silicon/robot/drone/swarm_items.dm]: > disintegrate

File: code/modules/mob/living/silicon/robot/drone/swarm_items.dm [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+		else
+			to_chat(user, "<span class='notice'>There is already a disintigration field affecting \the [T].</span>")
```
Comment (Anewbe) [inline code/modules/mob/living/silicon/robot/drone/swarm_items.dm]: > disintegrate

File: code/modules/mob/living/silicon/robot/drone/swarm_items.dm [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+		to_chat(user, "<span class='notice'>You deploy your decompiler and clear out the contents of \the [T].</span>")
+	else
+		to_chat(user, "<span class='danger'>Nothing on \the [T] is useful to you.</span>")
+	return
+
+/obj/effect/temporary_effect/pulse/disintigrate
+	name = "molecular debonding field"
+	desc = "This is something you do not want to near."
+	icon = 'icons/mob/swarmbot.dmi'
+	icon_state = "disintegrate_pulse"
```
Comment (Anewbe) [inline code/modules/mob/living/silicon/robot/drone/swarm_items.dm]: > disintegrate

File: code/modules/mob/living/silicon/robot/drone/swarm_items.dm [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+	light_range = 4
+	light_power = 5
+	light_color = "#00B4D9"
+	pulses_remaining = 5
+	pulse_delay = 2 SECONDS
+
+/obj/effect/temporary_effect/pulse/disintigrate/emp_act()
```
Comment (Anewbe) [inline code/modules/mob/living/silicon/robot/drone/swarm_items.dm]: > disintegrate

File: code/modules/mob/living/silicon/robot/drone/swarm_items.dm [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+	visible_message("<span class='warning'>\The [src] flickers, before dispersing energetically.</span>")
+	qdel(src)
+
+/obj/effect/temporary_effect/pulse/disintigrate/on_pulse()
```
Comment (Anewbe) [inline code/modules/mob/living/silicon/robot/drone/swarm_items.dm]: > disintegrate

File: code/modules/mob/living/silicon/robot/drone/swarm_items.dm [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+	var/turf/T = get_turf(src)
+	if(istype(T,/turf/simulated/wall))
+		explosion(get_turf(src), -1, -1, 1, 3, adminlog = 0)
+	else
+		qdel(src)
+
+/obj/effect/temporary_effect/pulse/disintigrate/Destroy()
```
Comment (Anewbe) [inline code/modules/mob/living/silicon/robot/drone/swarm_items.dm]: > disintegrate

File: code/modules/mob/living/silicon/robot/drone/swarm_items.dm [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+	name = "spectral projector"
+	desc = "A high-power laser gun capable of expelling concentrated gamma blasts, which are able to penetrate matter easier than \
+	standard xray beams, resulting in an effective 'anti-everything' energy weapon."
+	icon_state = "xray"
+	item_state = "xray"
+	origin_tech = list(TECH_COMBAT = 5, TECH_MATERIAL = 3, TECH_MAGNET = 2)
+	projectile_type = /obj/item/projectile/beam/shock
+	charge_cost = 175
+
+	self_recharge = 1
```
Comment (Anewbe) [inline code/modules/mob/living/silicon/robot/drone/swarm_items.dm]: TRUE/FALSE

File: code/modules/mob/living/silicon/robot/drone/swarm_items.dm [hunk 10]
```diff
@@ -0,0 +1,162 @@
...
+	use_external_power = 1
```
Comment (Anewbe) [inline code/modules/mob/living/silicon/robot/drone/swarm_items.dm]: TRUE/FALSE

File: code/modules/mob/living/silicon/robot/drone/swarm_items.dm [hunk 11]
```diff
@@ -0,0 +1,162 @@
...
+				if(plastic)
+					plastic.add_charge(750)
+			else
+				if(metal)
+					metal.add_charge(1000)
+		else
+			continue
+
+		qdel(W)
+		grabbed_something = 1
```
Comment (Anewbe) [inline code/modules/mob/living/silicon/robot/drone/swarm_items.dm]: TRUE/FALSE

File: code/modules/mob/living/silicon/robot/drone/swarm_items.dm [hunk 12]
```diff
@@ -0,0 +1,162 @@
...
+
+	if(!proximity) return //Not adjacent.
+
+	//We only want to deal with using this on turfs. Specific items aren't important.
+	var/turf/T = get_turf(target)
+	if(!istype(T))
+		return
+
+	//Used to give the right message.
+	var/grabbed_something = 0
```
Comment (Anewbe) [inline code/modules/mob/living/silicon/robot/drone/swarm_items.dm]: TRUE/FALSE

File: code/game/objects/effects/spiders.dm [hunk 13]
```diff
@@ -134,6 +134,8 @@
 	var/travelling_in_vent = 0
 	var/list/grow_as = list(/mob/living/simple_mob/animal/giant_spider, /mob/living/simple_mob/animal/giant_spider/nurse, /mob/living/simple_mob/animal/giant_spider/hunter)
 
+	var/stunted = FALSE
```
[the remark this answers is not in the document]
Comment (Mechoid) [reply code/game/objects/effects/spiders.dm]: Stunted determines if the spiderling grows into a smaller variant of the spider it chooses from grow_as.

File: code/modules/gamemaster/actions/spider_infestation.dm [hunk 14]
```diff
@@ -7,21 +7,28 @@
...
+/datum/gm_action/spider_infestation/set_up()
+	severity = pickweight(EVENT_LEVEL_MUNDANE = max(1,(12 - (3 * metric.count_people_in_department(ROLE_SECURITY)))),
+	EVENT_LEVEL_MODERATE = (7 + (2 * metric.count_people_in_department(ROLE_SECURITY))),
+	EVENT_LEVEL_MAJOR = (1 + (2 * metric.count_people_in_department(ROLE_SECURITY)))
+	)
+
+	switch(severity)
+		if(EVENT_LEVEL_MUNDANE)
+			spawntype = /obj/effect/spider/spiderling/stunted
+		if(EVENT_LEVEL_MODERATE)
```
[the remark this answers is not in the document]
Comment (Mechoid) [reply code/modules/gamemaster/actions/spider_infestation.dm]: Varied types are planned for the future, yes.

File: code/modules/detectivework/microscope/microscope.dm [hunk 15]
```diff
@@ -71,8 +71,8 @@
 			report.info += "Surface analysis has determined unique fingerprint strings:<br><br>"
 			for(var/prints in card.evidence)
 				report.info += "<span class='notice'>Fingerprint string: </span>"
-				if(!is_complete_print(prints))
-					report.info += "INCOMPLETE PRINT"
+				if(!is_complete_print(card.evidence[prints]))
+					report.info += "INCOMPLETE PRINT:[card.evidence[prints]]"
```
[the remark this answers is not in the document]
Comment (Mechoid) [reply code/modules/detectivework/microscope/microscope.dm]: This fixes fingerprints, yes.

File: code/game/objects/structures/ghost_pods/human.dm [hunk 16]
```diff
@@ -0,0 +1,246 @@
...
+
+	H.forceMove(T)
+
+	if(make_antag)
+		var/datum/antagonist/antag = all_antag_types[make_antag]
+		if(antag)
+			if(antag.add_antagonist(H.mind, 1, 1, 0, 1, 1))
+				log_admin("\The [src] made [key_name(src)] into a [antag.role_text].")
+
+	if(start_injured) // Done 3 different times to disperse damage.
```
[the remark this answers is not in the document]
Comment (Mechoid) [reply code/game/objects/structures/ghost_pods/human.dm]: To my knowledge it should refresh the appearance, I.E. wound visibility, but the damage should still be there.

**Pull request merged**